backend/corpora/common/upload.py

```diff
@@ -89,6 +90,8 @@ def upload(
                 dataset.reprocess()
 
     else:
+        if curator_tag and Dataset.get_dataset_from_curator_tag(db_session, collection_uuid, curator_tag):
```

Comment:
The API doc should be updated to reflect this behavior.

Reply:
will do

Reply:
This is the wrong behavior. If the tag already exists (i.e., the second clause of this statement evaluates to true), then we need to update/replace that dataset rather than throw an exception.

Reply:
@Bento007 tagging to make sure you see